code/game/gamemodes/wizard/wizard.dm

```diff
@@ -165,38 +173,138 @@
+					to_chat(wizard.current, "<span class='warning'>If there aren't any admins on and another wizard is camping you in the wizard lair, report them on the forums</span>")
+					message_admins("[wizard.current] went into crit in the wizard lair, another wizard is likely camping")
+					end_squabble(get_area(wizard.current))
+		if(wizard.current.client && wizard.current.client.is_afk() > 10 * 60 * 10) // 10 minutes
+			to_chat(wizard.current, "<span class='warning'><font size='4'>The Space Wizard Federation is upset with your performance and have terminated your employment.</font></span>")
+			wizard.current.gib() // Let's keep the round moving
+			continue
+		if(!wizard.current.client && !isskeleton(wizard.current))
+			continue // Could just be a bad connection, so SSD wiz's shouldn't be gibbed over it, but they're not "alive" either
 		wizards_alive++
 
-	if(!wizards_alive)
-		for(var/datum/mind/traitor in traitors)
-			if(!istype(traitor.current,/mob/living/carbon))
-				continue
-			if(traitor.current.stat==DEAD)
-				continue
-			traitors_alive++
+	if(wizards_alive && multi_wizard_drifting)
+		if(!time_checked)
+			time_checked = world.time
+		if(world.time > time_till_chaos && world.time > time_checked + delay_per_mage && (mages_made < wizard_cap))
+			time_checked = world.time
+			make_more_mages()
+	else if(!wizards_alive)
+		if(wizards.len >= wizard_cap)
+			finished = TRUE
+			return TRUE
+		else
+			make_more_mages()
+	return ..()
+
+// To silence all struggles within the wizard's lair
+/datum/game_mode/wizard/proc/end_squabble(var/area/wizard_station/A)
+	if(!istype(A)) return // You could probably do mean things with this otherwise
+	var/list/marked_for_death = list()
+	for(var/mob/living/L in A) // To hit non-wizard griefers
+		if(L.mind || L.client)
+			marked_for_death |= L
+	for(var/datum/mind/M in wizards)
+		if(istype(M.current) && istype(get_area(M.current), /area/wizard_station))
+			mages_made -= 1
+			wizards -= M // No, you don't get to occupy a slot
+			marked_for_death |= M.current
+	for(var/mob/living/L in marked_for_death)
+		if(L.stat == CONSCIOUS) // Probably a troublemaker - I'd like to see YOU fight when unconscious
+			to_chat(L, "<span class='userdanger'>STOP FIGHTING.</span>")
+		L.ghostize()
+		if(istype(L, /mob/living/carbon/brain))
+			// diediedie
+			var/mob/living/carbon/brain/B = L
+			if(istype(B.loc, /obj/item))
+				qdel(B.loc)
+			if(B && B.container)
+				qdel(B.container)
+		if(L)
+			qdel(L)
+	for(var/obj/item/spellbook/B in A)
+		// No goodies for you
+		qdel(B)
+
+/datum/game_mode/wizard/proc/make_more_mages()
+	if(making_mage || SSshuttle.emergency.mode >= SHUTTLE_ESCAPE)
+		return FALSE
+	making_mage = TRUE
+	var/list/candidates = list()
+	var/mob/dead/observer/harry = null
+	spawn(rand(200, 600))
+		message_admins("SWF is still pissed, sending another wizard.")
+		for(var/obj/item/I in summoned_items) //Wipe all summoned items so the new wiz isnt fucked over by a previous one.

```

Comment:
Not removing it from the list means it will never garbage collect.

Check your gosh darn references.